packages/cli/src/deploy/deploy.ts

```diff
@@ -51,6 +54,27 @@ export async function deploy<configInput extends ConfigInput>({
     resourceIds: [...tables.map((table) => table.tableId), ...systems.map((system) => system.systemId)],
   });
 
+  // deploy all dependent contracts, because system registration, module install, etc. all expect these contracts to be callable.
+  const contractTxs = (
+    await Promise.all([
+      ...uniqueBy(systems, (system) => system.address).map((system) =>
+        ensureContract({ client, bytecode: system.bytecode, label: `${resourceLabel(system)} system` })
+      ),
+      ...uniqueBy(config.modules, (mod) => mod.address).map((mod) =>
+        ensureContract({ client, bytecode: mod.bytecode, label: `${mod.name} module` })
+      ),
+    ])
+  ).flat();
+
+  if (contractTxs.length) {
+    debug("waiting for contracts");
+    // wait for each tx separately/serially, because parallelizing results in RPC errors
```

Comment:
> because parallelizing results in RPC errors

oh interesting

Reply:
I think because viem under the hood sets up a block watcher when you call `waitForTransaction`, so when we do this 100x in parallel, we get a LOT of polling of the RPC.

Since ultimately the `waitForTransactionReceipt` just does a `getTransactionReceipt` under the hood until it succeeds, it's fine to do this serially, because each transaction confirmed in the same block or blocks before will return immediately without the polling.

It'll ~resolve in the same amount of time with far less RPC calls.